src/routing.ts

```diff
@@ -30,9 +34,30 @@ export const initRouting = ({
     routing,
     hasCors: !!config.cors,
     onEndpoint: (endpoint, path, method, siblingMethods) => {
+      const requestType = endpoint.getRequestType();
+      if (
+        requestType === "json" &&
+        hasJsonIncompatibleSchema(endpoint.getSchema("input"), false)
```

Comment:
the hook can be called for the same `endpoint` reference multiple times when it's assigned to several routes, therefore it should be some optimization to reduce traversing

Reply:
addressed